src/internal/analytics-metadata/__tests__/utils.test.tsx

```diff
@@ -4,65 +4,81 @@
 import React from 'react';
 import { render } from '@testing-library/react';
 import { getGeneratedAnalyticsMetadata } from '../utils';
-import { METADATA_ATTRIBUTE } from '../attributes';
+import { METADATA_ATTRIBUTE, activateAnalyticsMetadata } from '../attributes';
 import { ComponentOne, ComponentThree } from './components';
 
 describe('getGeneratedAnalyticsMetadata', () => {
-  test('returns an empty object when input is null', () => {
-    expect(getGeneratedAnalyticsMetadata(null)).toEqual({});
-  });
-  test('skips malformed metadata', () => {
-    const { container } = render(
-      <div {...{ [METADATA_ATTRIBUTE]: "{'corruptedJSON':}" }}>
-        <ComponentOne malformed={true} />
-      </div>
-    );
-    const target = container.querySelector('#target') as HTMLElement;
-    expect(getGeneratedAnalyticsMetadata(target)).toEqual({
-      action: 'select',
-      detail: { label: 'event label', keyTwo: 'valueTwo' },
-      contexts: [
-        {
-          type: 'component',
-          detail: { name: 'ComponentOne', label: 'component label', properties: { multi: 'true' } },
-        },
-      ],
+  describe('with active analytics metadata', () => {
+    beforeAll(() => {
+      activateAnalyticsMetadata(true);
     });
-  });
-  test('resolves labels and merges metadata', () => {
-    const { container } = render(<ComponentOne />);
-    const target = container.querySelector('#target') as HTMLElement;
-    expect(getGeneratedAnalyticsMetadata(target)).toEqual({
-      action: 'select',
-      detail: { label: 'event label', keyOne: 'valueOne', keyTwo: 'overriddenValueTwo' },
-      contexts: [
-        {
-          type: 'component',
-          detail: { name: 'ComponentOne', label: 'component label', properties: { multi: 'true' } },
-        },
-      ],
+    test('returns an empty object when input is null', () => {
+      expect(getGeneratedAnalyticsMetadata(null)).toEqual({});
+    });
+    test('skips malformed metadata', () => {
+      const { container } = render(
+        <div {...{ [METADATA_ATTRIBUTE]: "{'corruptedJSON':}" }}>
```

Comment:
Are we here repeating the malformed cases? One on `<div />` and other one inside `<ComponentOne />` [here](https://github.com/cloudscape-design/component-toolkit/blob/3a71e8f32c2ff42b46bb8912b86fb95eb9077f35/src/internal/analytics-metadata/__tests__/components.tsx#L15)